### DISCUSSION TITLE
Plan on next release?

### DISCUSSION THREAD

**Discussion by ChaseLewis** [ash-rs/ash#882] [Q&A] [answered]
Noticed there hasn't been a release in nearly a year despite quite a few Vulkan releases. Mainly was wanting to try to mess with RTX IO released last July, but extensions related to it aren't in ash. Noticed there have been many updates to the repo, but no new release. Curious what the status is?

Relevant Extensions
VK_NV_memory_decompression
VK_NV_copy_memory_indirect

**Comment by Friz64** [comment]
#787 still applies. I think we also want to get #799 in. I don't know about #828 and #664 but they also have a https://github.com/ash-rs/ash/labels/Blocking%20a%20release label.

> VK_NV_copy_memory_indirect

This extension does not have a high level wrapper yet. This would need to be contributed.

**Reply by MarijnS95** [reply]
Both those extensions were released in Vulkan `1.3.233`.  Given that the latest `ash` releases have been made at `.235`, `.238` and `.251` respectively, at least the structures, raw function signatures and extension loaders are available for quite some time:

https://docs.rs/ash/latest/ash/vk/struct.NvMemoryDecompressionFn.html
https://docs.rs/ash/latest/ash/vk/struct.NvCopyMemoryIndirectFn.html

A high-level wrapper has only been written for `VK_NV_memory_decompression` in https://github.com/ash-rs/ash/pull/761 and will show up in the next release.

High-level hand-written/**curated** extension wrappers have not yet been written for `VK_NV_copy_memory_indirect` (and many others), and wouldn't have shown up in the next release anyway.  Now that there's a request for them, I'll add it to the list at https://github.com/ash-rs/ash/issues/178 and make sure they get wrapped after landing some extension-breaking changes first.

---

As a request about the next release has already been posted and is already tracked in https://github.com/ash-rs/ash/issues/787, I'll close this issue so that anyone subscribed there remains in the loop of any followup.  It's quite hard for both maintainers and interested users to track information across duplicate issues/discussions otherwise.

In general the plan is still to ensure at least all PRs marked with https://github.com/ash-rs/ash/labels/Blocking%20a%20release are merged in, to reduce the number of known breaking changes we'd have to fold into the next release otherwise. At least we've made big strides towards the big blockers very recently.

---

Note that this reply used to be much more elaborate, but GitHub ate the prepared message when editing and saving the broken label link in Friz's post. Forgive me if I left out something important.

**Reply by chase-lewis-akasa** [reply]
Thanks, didn't realize how to access non curated extensions. All the examples had always used curated so was unaware of how some of that worked in ash.

**Comment by MarijnS95** [comment, marked as answer]
The release is now out!

https://github.com/ash-rs/ash/releases/tag/0.38.0
